docs/03-onchain-architecture.md

```diff
@@ -0,0 +1,94 @@
+# 3. Smart Contract Architecture
```

Comment:
I would add a table (section?) here which clearly maps each contract name with their address (rather than an inline link) and perhaps a note, e.g.

| Contract | Address | ENS | Note| 
|---|---|---|---|
| Vesting | 0x... | vesting.pg.eth |  **Default donation address for Protocol Guild** |
| PassThrough | ... | ... | Used by DAO to redirected vested funds in case of malicious donations | |
....

Reply:
We should also have a second table with all the L2 addresses!